app/Filament/Resources/ServerResource/Pages/EditServer.php

```diff
@@ -460,6 +492,66 @@ public function form(Form $form): Form
                                     ])
                                     ->required(),
 
+                                Forms\Components\TagsInput::make('ports')
+                                    ->columnSpan(3)
+                                    ->placeholder('Example: 25565, 8080, 1337-1340')
+                                    ->splitKeys(['Tab', ' ', ','])
+                                    ->helperText(new HtmlString('
+                                        These are the ports that users can connect to this Server through.
+                                        <br />
+                                        You would typically port forward these on your home network.
+                                    '))
+                                    ->label('Ports')
+                                    ->formatStateUsing(fn (Server $server) => $server->ports->map(fn ($port) => (string) $port)->all())
+                                    ->afterStateUpdated(self::ports(...))
+                                    ->live(),
+
+                                Forms\Components\Repeater::make('portVariables')
```

Comment:
If there are no port variables we should display a placeholder/ info text instead of just empty whiteness.
![grafik](https://github.com/user-attachments/assets/42221f4c-0ccc-414e-bb7a-cd05b9c98da4)